### PR TITLE
Update UI to display contact info for user

### PR DESCRIPTION
**Pull request by willdoran**
…nd messages

This pull request makes the following changes:
- Adds contact info to user
- Displays contact info for posts metadata
- Displays contact info for "Conversation with Author" section
- Display contact info on User list page
- Displays contact info on User edit page

Testing checklist:
## User section
- [x] On the User list view, beneath the User's name you should see their Role and any Contacts they have displayed in a vertical column
screen shot 2017-12-13 at 10 38 35

- [x] On the User list view it should be possible to search via contact; for example phone number

- [x] On the User edit page you should see a new section called Contacts that shows any available contacts
screen shot 2017-12-13 at 10 48 18

- [x] The contact field should not be editable
- [x] You can update and save the User without issue
- [x] The contact section should not appear for users you have no contacts
- [x] The contact section should not appear when creating a new User

## Map view/Data Mode
- [x] For Posts created from inbound data sent to the Platform from the User's contact
  - [x] The author meta data on the Post Cards should show the User's name

## Post detail view
- [x] On a Post created from inbound data sent to the Platform from the user's contact
  - [x] The User's name should appear on the Conversation with Author section *not* the contact number

- [x] I certify that I ran my checklist

Fixes ushahidi/platform# .

Relies on [ushahidi/platform/pull/2359](https://github.com/ushahidi/platform/pull/2359)

Ping @ushahidi/platform

### CODE REVIEW

File: app/main/posts/detail/post-messages.directive.js [hunk 1]
```diff
@@ -51,10 +52,14 @@ function (
                 // Initialize
                 if ($scope.post.contact.contact) {
```
Comment (rowasc) [inline app/main/posts/detail/post-messages.directive.js]: @willdoran maybe `$scope.post.contact && $scope.post.contact.contact ` would be safer in the off chance that post.contact  is not there ? Not sure if that's even possible (not being there), just asking

Comment (rowasc) [reply app/main/posts/detail/post-messages.directive.js]: ah ok nvm, just realized you are using the contact.id in the else so I'm assuming it's going to be available. Sorry

Comment (willdoran) [reply app/main/posts/detail/post-messages.directive.js]: done

**Comment by rowasc** [comment]
@willdoran looks good, passed testing! Only 1 thing that I don't think it's from this PR, but maybe you can check: the user email when you have a phone number (like the test "Robbie" user) is  pushed to the left (it actually happens to the third item no matter what the content is.) 

Screenshot: 
screen shot 2017-12-14 at 3 27 41 pm

**Pull request merged**